Map/RoomBase.gd

```diff
@@ -0,0 +1,10 @@
+extends Resource
+class_name RoomBase
+## Class for a room on the map. Provides base classes for all rooms to use regardless of the type
+##
+## Holds basic functionality for a room on the map
+
+@export var possible_events: Array = [EventHeal, EventMob, EventRandom, EventShop]
```

Comment:
`EventBase` has one but you can't reference it directly. A way to do that is to create a new `EventBase` and then do a `event.EVENT_CLASSIFICATION`

A better way is to use a global file to store our `EVENT_CLASSIFICATION`, which extends from a Node and to auto-load it at project start with (Project -> Project settings -> Auto-load). If we call this script `GLOBAL_VAR.gd` you can reference his elements with `GLOBAL_VAR.ELEMENT`.
I think we should implement this second option because it might be useful in the future. I should have thought about loading in other files when I first asked for this array...

Reply:
> **A better way** is to use a global file to store our `EVENT_CLASSIFICATION`, which extends from a Node and to auto-load it at project start with (Project -> Project settings -> Auto-load). If we call this script `GLOBAL_VAR.gd` you can reference his elements with `GLOBAL_VAR.ELEMENT`. I think we should implement this second option because it might be useful in the future. I should have thought about loading in other files when I first asked for this array...
for this, did you want it to be added in this pr, or a separate one?

Reply:
You can do that in the current PR since it's quick to do and you need the list of events (which already exists).